akka-http-core/src/main/scala/akka/http/javadsl/ConnectionContext.scala

```diff
@@ -23,7 +24,12 @@ object ConnectionContext {
             enabledProtocols: Optional[JCollection[String]],
             clientAuth: Optional[TLSClientAuth],
             sslParameters: Optional[SSLParameters]) =
-    scaladsl.ConnectionContext.https(sslContext, sslParameters = OptionConverters.toScala(sslParameters))
+    scaladsl.ConnectionContext.https(
+      sslContext,
+      OptionConverters.toScala(enabledCipherSuites).map(_.asScala.toList),
+      OptionConverters.toScala(enabledProtocols).map(_.asScala.toList),
+      OptionConverters.toScala(clientAuth),
+      OptionConverters.toScala(sslParameters))
```

Comment:
Well, we really need regression tests anyway.

-- 
Cheers,
√
On Apr 26, 2016 6:37 PM, "Konrad Malawski" <notifications@github.com> wrote:

> In akka-http-core/src/main/scala/akka/http/javadsl/ConnectionContext.scala
> <https://github.com/akka/akka/pull/20399#discussion_r61119525>:
>
> > @@ -23,7 +24,12 @@ object ConnectionContext {
> >              enabledProtocols: Optional[JCollection[String]],
> >              clientAuth: Optional[TLSClientAuth],
> >              sslParameters: Optional[SSLParameters]) =
> > -    scaladsl.ConnectionContext.https(sslContext, sslParameters = OptionConverters.toScala(sslParameters))
> > +    scaladsl.ConnectionContext.https(
> > +      sslContext,
> > +      OptionConverters.toScala(enabledCipherSuites).map(_.asScala.toList),
> > +      OptionConverters.toScala(enabledProtocols).map(_.asScala.toList),
> > +      OptionConverters.toScala(clientAuth),
> > +      OptionConverters.toScala(sslParameters))
>
> I can add a test, sure... covering all delegating methods from java ->
> scala with tests though is not really possible (unless we get a person
> full-time on testing that..)
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/akka/akka/pull/20399/files/fffdb4baa56169299e3fb913aa9fd45958dbc988#r61119525>
>